pkg/k8s/endpoints.go

```diff
@@ -363,7 +363,7 @@ func SupportsEndpointSlice() bool {
 // SupportsEndpointSliceV1 returns true if cilium-operator or cilium-agent should
 // watch and process endpoint slices V1.
 func SupportsEndpointSliceV1() bool {
-	return version.Capabilities().EndpointSliceV1 && option.Config.K8sEnableK8sEndpointSlice
+	return SupportsEndpointSlice() && version.Capabilities().EndpointSliceV1
```

Comment:
My reading is that `SupportsEndpointSlice()` checks for < k8s 1.21 EndpointSlice support. Is it guaranteed that on >= k8s 1.21 not only `Capabilities().EndpointSliceV1`, but also `Capabilities().EndpointSlice` will be set to `true` too?

Reply:
The idea was to have the option `option.Config.K8sEnableK8sEndpointSlice` used to disable EndpointSlice support for both v1beta1 and v1.

Reply:
Yes, and it will:

```
func SupportsEndpointSlice() bool {
	return version.Capabilities().EndpointSlice && option.Config.K8sEnableK8sEndpointSlice
}
```

It checks the option and now this PR simply combines the two functions. As mentioned, one variant cannot return true and the other false, and vice versa.